docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+    the names ``extension`` and ``type``. The ``extension`` is
+    required and the value must be a URI that identifies the
+    extension and dereferences to a human-readable representation of
+    the specification.  The ``type`` is required and the value is
+    defined by the extension.
+
+``chunk_memory_layout``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+    The internal memory layout of the chunks. Use the value "C" to
+    indicate `C contiguous memory layout`_ or "F" to indicate
+    `F contiguous memory layout`_ as defined in this specification.
+
+    The ``chunk_memory_layout`` value is an extension point and may be
+    defined by an extension. If the chunk memory layout type
+    is defined by an extension, then the value must be an
+    object containing the names ``extension`` and ``type``. The
+    ``extension`` is required and the value must be a URI that
+    identifies the extension and dereferences to a
+    human-readable representation of the specification.  The ``type`` is
+    required and the value is defined by the extension.
+
+``fill_value``
+^^^^^^^^^^^^^^
+
+    Provides an element value to use for uninitialised portions of the
+    Zarr array.
+
+    If the data type of the Zarr array is Boolean then the value must
+    be the literal ``false`` or ``true``. If the data type is one of
+    the integer data types defined in this specification, then the
+    value must be a number with no fraction or exponent part and must
+    be within the range of the data type.
+
+    For any data type, if the ``fill_value`` is the literal ``null``
+    then the fill value is undefined and the implementation may use
+    any arbitrary value that is consistent with the data type as the
+    fill value.
+
+    If the ``data_type`` of an array is defined in a ``data_type`` extension,
+    then said extension is responsible for interpreting the value of
+    ``fill_value`` and return a suitable type that can be used.
+
+    For core data types for which fill values are not permitted in JSON or
+    for which decimal representation could be lossy, a string representing of
+    the binary (starting with ``0b``) or hexadecimal value (starting with
+    ``0x``) is accepted. This string must include all leading or trailing
+    zeroes necessary to match the given type size. The string values ``"NaN"``,
+    ``"+Infinity"`` and ``"-Infinity"`` are also understood for floating point
+    data types.
+
+``extensions``
+^^^^^^^^^^^^^^
+
+    See the top level metadata extension section for the time being.
+
+
+``attributes``
+^^^^^^^^^^^^^^
+
+    The value must be an object. The object may contain any name/value
+    pairs. Intended to allow storage of arbitrary user metadata
+
+
+  .. note::
+    The question of whether core metadata and user attributes should be
+    stored together or in separate documents is a topic of ongoing discussion.
+    (See https://github.com/zarr-developers/zarr-specs/issues/72.)
```

Comment:
It would also be possible to accomplish this as an extension. The default could be to have attributes in the main document. An extension could add "attribute_documents" pointing at other files.

Reply:
#72 is tagged as a v3 core issue and therefore meant to be resolved for v3. Let's discuss this there directly.